crates/macro-support/src/lib.rs

```diff
@@ -104,9 +104,70 @@ pub fn expand_class_marker(
     Ok(tokens)
 }
 
+/// Describes how to rename the fields, methods, and classes in the generated
+/// bindings.
+///
+/// This design heavily mimics the `#[serde(rename_all = "...")]` attribute.
+#[derive(Debug, Clone, Copy)]
+enum RenameRule {
+    None,
+    CamelCase,
+}
+impl RenameRule {
+    fn snake_case_to_camel_case(name: String) -> String {
+        let mut camel = String::new();
+        let mut capitalize = false;
+        for c in name.chars() {
+            if c == '_' {
+                capitalize = true;
+            } else if capitalize {
+                camel.push(c.to_ascii_uppercase());
```

Comment:
```suggestion
                camel.push(c.to_uppercase());
```
I'm not sure if we even allow non-ASCII symbols.

Reply:
We don't.

```rs
#[wasm_bindgen]
pub fn nöp() {}
```
```
        error[E0754]: `#[no_mangle]` requires ASCII identifier
         --> C:\Users\micha\Git\wasm-bindgen\crates\cli\tests/reference\nop.rs:3:1
          |
        3 | #[wasm_bindgen]
          | ^^^^^^^^^^^^^^^
          |
          = note: this error originates in the attribute macro `wasm_bindgen` (in Nightly builds, run with -Z macro-backtrace for more info)

        For more information about this error, try `rustc --explain E0754`.
        error: could not compile `reference-test` (lib) due to 1 previous error

```

Since I know a bit about how WBG handles name mangling because of #4230, I thought about supporting arbitrary strings as JS names after that PR is through. This would naturally add support for non-ASCII identifiers as well.

But I'm not sure whether this means we should use `to_ascii_uppercase` or `to_uppercase`. What do you think?